module/x/gravity/types/interfaces.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	proto "github.com/gogo/protobuf/proto"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+)
+
+// EthereumSignature represents one validtors signature for a given
+// outgoing ethereum transaction
+type EthereumSignature interface {
+	proto.Message
+
+	GetSigner() common.Address
+	GetSignature() hexutil.Bytes
+	GetStoreIndex() []byte
+	Validate() error
+}
+
+// _ EthereumSignature = &ContractCallTxSignature{}
+// _ EthereumSignature = &BatchTxSignature{}
+// _ EthereumSignature = &SignerSetUpdateTxSignature{}
+
+// EthereumEvent represents a event from the gravity contract
+// on the counterparty ethereum chain
+type EthereumEvent interface {
+	proto.Message
```

Comment:
What does this notation do?

Reply:
https://golang.org/doc/effective_go#embedding This interface embeds the `proto.Message` interface as well.